packages/remix-react/components.tsx

```diff
@@ -360,7 +360,7 @@ export function Links() {
 
         return (
           <link
-            key={link.rel + (link.href || "") + (imageSrcSet || "")}
+            key={key}
```

Comment:
They key is now the same value used for deduping rather than a custom subset of properties. To enable this, the utilities that return arrays of link descriptors to render now provide `Keyed` variants that expose a `key` property for each link. This ensures we never get any duplicate key errors, whereas the previous logic could result in duplicate keys if `rel`, `href` and/or `imageSrcSet` properties were the same but other properties differed.